diff --git a/src/main.cpp b/src/main.cpp
index 1b44331..81894f4 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -70,6 +70,12 @@ int fUseUPnP = false;
 
 CHooks* hooks;
 
+void runCommand(std::string strCommand)
+{
+    int nErr = ::system(strCommand.c_str());
+    if (nErr)
+        printf("runCommand error: system(%s) returned %d\n", strCommand.c_str(), nErr);
+}
 
 
 
@@ -1190,6 +1196,14 @@ bool CBlock::SetBestChain(CTxDB& txdb, CBlockIndex* pindexNew)
     nTransactionsUpdated++;
     printf("SetBestChain: new best=%s  height=%d  work=%s\n", hashBestChain.ToString().substr(0,20).c_str(), nBestHeight, bnBestChainWork.ToString().c_str());
 
+    std::string strCmd = GetArg("-blocknotify", "");
+
+    if (!IsInitialBlockDownload() && !strCmd.empty())
+      {
+        boost::replace_all(strCmd, "%s", hashBestChain.GetHex());
+        boost::thread t(runCommand, strCmd); // thread runs free
+      }
+
     return true;
 }
 
